File: androidApp/src/main/java/com/vasanth/kmm/apparchitecture/android/ui/userlist/components/sampledata/UserListSampleData.kt
package com.vasanth.kmm.apparchitecture.android.ui.userlist.components.sampledata

import com.vasanth.kmm.apparchitecture.presentation.userlist.model.UserListUIState
import com.vasanth.kmm.apparchitecture.presentation.userlist.model.UserUIModel

object UserListSampleData {

    fun createUserUIModel(id: Long): UserUIModel {
        val user = UserUIModel(
            id = id,
            name = "Vasanth",
            email = "test@email.com",
            avatar = "https://picsum.photos/200/300"
        )
        return user
    }

    fun createUserUIModels(): List<UserUIModel> {
        val users = List(5) { index ->
            createUserUIModel(id = index.toLong())
        }
        return users
    }

    fun createDataUIState(): UserListUIState.Data {
        val state = UserListUIState.Data(
            users = createUserUIModels()
        )

        return state
    }
}

File: iosApp/iosApp/Source/UI/Userlist/View/Components/SampleData/UserListSampleData.swift
//
//  UserListSampleData.swift
//  iosApp
//
//  Created by Vasanthakumar Annadurai on 06/09/23.
//

import Foundation
import shared

struct UserListSampleData {
    
    static func createUserUIModel(id: Int64) -> UserUIModel {
        let user = UserUIModel(
            id: id,
            name: "Vasanth",
            email: "test@email.com",
            avatar: "https://picsum.photos/200/300"
        )
        return user
    }
    
    static func createUserUIModels() -> [UserUIModel] {
        var users = [UserUIModel]()
        for id in 0..<5 {
            users.append(createUserUIModel(id: Int64(id)))
        }
        return users
    }
    
    static func createDataUIState() -> UserListUIStateData {
        let state = UserListUIStateData(
            users: createUserUIModels()
        )
        return state
    }
}
